### PR TITLE
Spark 3.4: Action to compute table stats

### PR DESCRIPTION
**Pull request by karuppayya**
Backport of https://github.com/apache/iceberg/pull/10288

cc: @aokolnychyi @szehon-ho

### CODE REVIEW

**Review by singhpk234 — APPROVED** [review]

File: spark/v3.3/build.gradle [hunk 1]
```diff
@@ -59,6 +59,7 @@ project(":iceberg-spark:iceberg-spark-${sparkMajorVersion}_${scalaVersion}") {
     implementation project(':iceberg-parquet')
     implementation project(':iceberg-arrow')
     implementation("org.scala-lang.modules:scala-collection-compat_${scalaVersion}:${libs.versions.scala.collection.compat.get()}")
+    implementation("org.apache.datasketches:datasketches-java:${libs.versions.datasketches.get()}")
```
Comment (huaxingao) [inline spark/v3.3/build.gradle:62]: Do we need to change v3.3 `build.gradle`?

File: spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/actions/NDVSketchUtil.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  private static Row computeNDVSketches(
+      SparkSession spark, Table table, Snapshot snapshot, List<String> colNames) {
+    return spark
+        .read()
+        .format("iceberg")
+        .option(SparkReadOptions.SNAPSHOT_ID, snapshot.snapshotId())
+        .load(table.name())
+        .select(toAggColumns(colNames))
+        .first();
```
Comment (dramaticlly) [inline spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/actions/NDVSketchUtil.java:82]: do we need backport #10984 for spark 3.4 as well per Anton's comment in https://github.com/apache/iceberg/pull/10288/files#r1726000959? Happy to help

**Review by dramaticlly — APPROVED** [review]
LGTM, looks like CI run into some transient test failure

**Review by szehon-ho — APPROVED** [review]

**Comment by szehon-ho** [comment]
Merged, thanks @karuppayya and all for additional review.

File: spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/actions/NDVSketchUtil.java (hunk 2 above)
Comment (tedyu) [inline spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/actions/NDVSketchUtil.java:82]: should we consider calling `.cache()` before `.first()` ?

**Pull request merged**